spec.html

```diff
@@ -36148,11 +36052,21 @@ <h1>CreateMapIterator ( _map_, _kind_ )</h1>
         <p>The abstract operation CreateMapIterator takes arguments _map_ and _kind_. This operation is used to create iterator objects for Map methods that return such iterators. It performs the following steps when called:</p>
         <emu-alg>
           1. Perform ? RequireInternalSlot(_map_, [[MapData]]).
-          1. Let _iterator_ be OrdinaryObjectCreate(%MapIteratorPrototype%, &laquo; [[IteratedMap]], [[MapNextIndex]], [[MapIterationKind]] &raquo;).
-          1. Set _iterator_.[[IteratedMap]] to _map_.
-          1. Set _iterator_.[[MapNextIndex]] to 0.
-          1. Set _iterator_.[[MapIterationKind]] to _kind_.
-          1. Return _iterator_.
+          1. Let _closure_ be a new Abstract Closure with no parameters that captures _map_ and _kind_ and performs the following steps when called:
+            1. Let _entries_ be the List that is _map_.[[MapData]].
+            1. Let _index_ be *+0*.
+            1. Repeat, while _index_ is less than the number of elements of _entries_,
+              1. Let _e_ be the Record { [[Key]], [[Value]] } that is the value of _entries_[_index_].
+              1. Set _index_ to _index_ + 1.
+              1. If _e_.[[Key]] is not ~empty~, then
+                1. If _kind_ is ~key~, let _result_ be _e_.[[Key]].
+                1. Else if _kind_ is ~value~, let _result_ be _e_.[[Value]].
+                1. Else,
+                  1. Assert: _kind_ is ~key+value~.
+                  1. Let _result_ be ! CreateArrayFromList(&laquo; _e_.[[Key]], _e_.[[Value]] &raquo;).
+                1. Perform ? Yield(_result_).
+            1. Return *undefined*.
+          1. Return ? CreateBuiltinGeneratorInstance(_closure_, ~Map Iterator~, %MapIterator.prototype%).
```

Comment:
```suggestion
          1. Return ? CreateBuiltinGeneratorInstance(_closure_, ~Map Iterator~, %MapIteratorPrototype%).
```

Reply:
But @jmdyck is suggesting %XIterator.prototype% in the previous review

Reply:
@Jack-Works anywhere `XIterator` and `XIteratorPrototype` are *both* intrinsics, we should prefer `XIterator.prototype`. In this case, only the non-dotted form exists.